src/lib.rs

```diff
@@ -0,0 +1,631 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use async_trait::async_trait;
+use prost::Message;
+use std::collections::HashMap;
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc, Mutex,
+};
+use std::time::Duration;
+use uprotocol_sdk::{
+    rpc::{RpcClient, RpcClientResult, RpcMapperError, RpcServer},
+    transport::{datamodel::UTransport, validator::Validators},
+    uprotocol::{
+        Data, UAttributes, UCode, UEntity, UMessage, UMessageType, UPayload, UPayloadFormat,
+        UPriority, UStatus, UUri, Uuid,
+    },
+    uri::{
+        serializer::{MicroUriSerializer, UriSerializer},
+        validator::UriValidator,
+    },
+};
+use zenoh::{
+    config::Config,
+    prelude::{r#async::*, Sample},
+    queryable::{Query, Queryable},
+    sample::AttachmentBuilder,
+    subscriber::Subscriber,
+};
+
+pub struct ZenohListener {}
+pub struct ULinkZenoh {
```

Comment:
ULinkZenoh needs to construct its own rpc response topic that it has to send to the server so that the responding uE knows who sent the request (in the RpcServer response listener).